bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/internal/thing/ThingTypeResource.java

```diff
@@ -164,25 +160,6 @@ public Response getByUID(@PathParam("thingTypeUID") @ApiParam(value = "thingType
         }
     }
 
-    @GET
-    @Path("/{thingTypeUID}/firmwares")
-    @Produces(MediaType.APPLICATION_JSON)
-    @ApiOperation(value = "Get all available firmwares for provided thingType", response = StrippedThingTypeDTO.class, responseContainer = "Set")
-    @ApiResponses(value = { @ApiResponse(code = 200, message = "OK"),
-            @ApiResponse(code = 204, message = "No firmwares found.") })
-    public Response getFirmwares(@PathParam("thingTypeUID") @ApiParam(value = "thingTypeUID") String thingTypeUID,
```

Comment:
With the drop of this method you can also remove the entire dependency to the `FirmwareRegistry`, can't you?

Reply:
Yes, you are right. Thanks for noticing that.